code/game/objects/items/weapons/cigs_lighters.dm

```diff
@@ -248,57 +356,60 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 	if(isliving(loc))
 		M.IgniteMob()
 	smoketime--
-	if(smoketime < 1)
-		new type_butt(location)
-		processing_objects.Remove(src)
+	if(smoketime <= 0)
+		new type_butt(location) //Spawn the cigarette butt
+		lit = 0 //Actually unlight the cigarette so that the lighting can update correctly
+		update_brightness()
 		if(ismob(loc))
 			M << "<span class='notice'>Your [name] goes out.</span>"
-			M.u_equip(src,0)	//un-equip it so the overlays can update
+			M.u_equip(src, 0)	//Un-equip it so the overlays can update
 			M.update_inv_wear_mask(0)
-		del(src)
+		qdel(src)
 		return
 	if(location)
-		location.hotspot_expose(700, 5,surfaces=istype(loc,/turf))
-	if(reagents && reagents.total_volume)	//	check if it has any reagents at all
-		if(iscarbon(M) && (src == M:wear_mask)) // if it's in the human/monkey mouth, transfer reagents to the mob
+		location.hotspot_expose(700, 5, surfaces = istype(loc, /turf))
+	//Oddly specific and snowflakey reagent transfer system below
+	if(reagents && reagents.total_volume)	//Check if it has any reagents at all
+		if(iscarbon(M) && (src == M.wear_mask)) //If it's in the human/monkey mouth, transfer reagents to the mob
 			if(M.reagents.has_reagent("lexorin") || M_NO_BREATH in M.mutations || istype(M, /obj/machinery/atmospherics/unary/cryo_cell))

```

Comment:
There's an error on this line you should fix. It should be
`istype(M.loc, /obj/machinery/atmospherics/unary/cryo)`